torch/_ops.py

```diff
@@ -243,19 +244,27 @@ def dispatch(self, dispatch_key, *args, **kwargs):
             return dispatch_functorch(self, args, kwargs)
 
         if dispatch_key == torch._C.DispatchKey.Python:
-            # TODO(voz): We should walk all the nodes here / turn it into a list, topmode is ok for now.
+            # The place to handle ProxyTorchDispatchMode, FakeTensorMode, etc
+            from torch.utils._python_dispatch import _pop_mode_temporarily
+
             curr_mode = _get_current_dispatch_mode()
             assert (
                 curr_mode is not None
             ), "Illegal invocation of dispatch on torch._C.DispatchKey.Python without a mode."
             assert (
                 type(curr_mode) in self.python_key_mode_table
             ), f"Current active mode {curr_mode} not registered"
-            # TODO(voz): The idea behind this is that we do not yet support dispatch by key + mode, only key.
-            return self.python_key_mode_table[type(curr_mode)](*args, **kwargs)
+            handler = self.python_key_mode_table[type(curr_mode)]
+            if isinstance(curr_mode, ProxyTorchDispatchMode):
+                with _pop_mode_temporarily() as mode:
+                    args = (mode,) + args
+                    return handler(*args, **kwargs)
+            else:
+                return handler(*args, **kwargs)
```

Comment:
We should unconditionally _pop_mode_temporarily() for all modes, not just ProxyTorchDispatchMode. Does this run you into problems?